### ISSUE TITLE
Using deprecated methods

### ISSUE THREAD

**Issue by tijs** [mopinion/mopinion-sdk-react-native#15]
It seems the the Mopinion code (and/or the ModalBox dependency it uses) uses deprecated lifecycle methods. These should be removed/refactored before the next React release where they   will no longer be available.

From the Metro logs:

```
 WARN  Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move code with side effects to componentDidMount, and set initial state in the constructor.
* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: ModalBox, Mopinion
```

**Comment by stevenkuipers** [comment]
Hi Tijs, 

Thanks for letting us know!
We are aware of this and will update this in a future release.

**Comment by tijs** [comment]
@stevenkuipers any news on this?

**Comment by tijs** [comment]
@stevenkuipers @mopinion again... any news on this. Still no new release.

Please update the following components: ModalBox, Mopinion
 WARN  Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: ModalBox, Mopinion

**Comment by Anwardo** [comment]
Just released v1.2.2 which fixes this issue!
